app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1365,17 +1404,13 @@ public void onClick(View v) throws NoSuchElementException {
     // We don't need to set or check the dialogId since it's only used for that single dialog
     @Override
     public void onColorSelected(int dialogId, int color) {
+        // Save new colour
+        updateTempState(LoyaltyCardField.headerColor, color);
+
         // Unset image if set
-        setCardImage(thumbnail, null, false);
+        setThumbnailImage(null);
         mIconRemoved = true;
         mIconUnsaved = false;
```

Comment:
Just a minor inconsistency with other code where the booleans are assigned before the call.

Reply:
```suggestion
        mIconRemoved = true;
        mIconUnsaved = false;
        setThumbnailImage(null);
```